File: Perfect Numbers - GFG/perfect-numbers.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  public:
    int isPerfectNumber(long long n) {
        // code here
        
        long long ans=0;  //initialize the ans as 0
        
        for (int i=1;i<=sqrt(n);i++){   //run the loop upto sqrt(n)
            
            if(n%i==0){
                ans+=i;       //add the fatctor
                
                if(i!=(n%i)){   //check wheathe the factor is not sqrt(n)
                
                    ans+=(n/i);   //add anothe factor
                }
            }
        }
        
        // now ans is the sum of all unique factor of ans 
        
        
        
        // check wheathe rthe given no. is not 1 or perect 
        if((ans-n)==n and n!=1){
            return 1;
        }
        else{
            return 0;
        }
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        long long N;
        
        cin>>N;

        Solution ob;
        cout << ob.isPerfectNumber(N) << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Perfect Numbers - GFG/perfect-numbers.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            long N = Long.parseLong(read.readLine());

            Solution ob = new Solution();
            System.out.println(ob.isPerfectNumber(N));
        }
    }
}
// } Driver Code Ends



//User function Template for Java

class Solution {
    static int isPerfectNumber(long N) {
        
        
        // code here
        
         long ans = 0; // Initialize ans as 0
        
        for (int i = 1; i <= Math.sqrt(N); i++) {
            if (N % i == 0) {
                ans += i; // Add the factor
                
                if (i != (N / i)) { // Check whether the factor is not sqrt(N)
                    ans += (N / i); // Add another factor
                }
            }
        }
        
        // Now, ans is the sum of all unique factors of N
        
        // Check whether the given number is not 1 and is perfect
        if ((ans - N) == N && N != 1) {
            return 1;
        } else {
            return 0;
        }
        
        
        
    }
};